### PR TITLE
#213 - Second order linear systems

### LINKED ISSUE #213: Add second order linear system
Add a struct to hold systems of the form `Mx''(t) + Cx'(t) + Kx(t) = Bu(t)` as a `SecondOrderLinearContinuousSystem`.

cc: @dfcaporale, @jorgepz.

### PR DESCRIPTION
**Pull request by mforets**
Closes #213.

Tests are still missing.

### CODE REVIEW

**Comment by mforets** [comment]
This is ready from my side, tests pass locally. 

In principle we could make the transformation from second order a system to a first order one as a new method in this library. For the time being that's being done in ReachabiltyAnalysis.jl.

File: src/systems.jl [hunk 1]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
 """
 NoisyConstrainedBlackBoxControlDiscreteSystem
+
+@doc """
+    SecondOrderLinearContinuousSystem
+
+Continuous-time second order linear system of the form:
+
+```math
+    Mx(t)''(t) + Cx'(t) + Kx(t) = 0\\; \\forall t.
```
Comment (schillic) [inline src/systems.jl:1705]:
```suggestion
    Mx''(t) + Cx'(t) + Kx(t) = 0\\; \\forall t.
```
I guess?

File: src/systems.jl [hunk 2]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
+    end
+end
+
+@doc """
+    SecondOrderAffineContinuousSystem
+
+Continuous-time second order affine system of the form:
+
+```math
+    Mx(t)''(t) + Cx'(t) + Kx(t) = b\\; \\forall t.
```
Comment (schillic) [inline src/systems.jl:1779]:
```suggestion
    Mx''(t) + Cx'(t) + Kx(t) = b\\; \\forall t.
```

File: src/systems.jl [hunk 3]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
+    end
+end
+
+@doc """
+    SecondOrderConstrainedLinearControlContinuousSystem
+
+Continuous-time second order constrained linear control system of the form:
+
+```math
+    Mx(t)''(t) + Cx'(t) + Kx(t) = Bu(t), \\; x(t) ∈ \\mathcal{X}, \\; u(t) ∈ \\mathcal{U} \\; \\forall t.
```
Comment (schillic) [inline src/systems.jl:1858]:
```suggestion
    Mx''(t) + Cx'(t) + Kx(t) = Bu(t), \\; x(t) ∈ \\mathcal{X}, \\; u(t) ∈ \\mathcal{U} \\; \\forall t.
```

File: src/systems.jl [hunk 4]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
+    end
+end
+
+@doc """
+    SecondOrderConstrainedAffineControlContinuousSystem
+
+Continuous-time second order constrained affine control system of the form:
+
+```math
+    Mx(t)''(t) + Cx'(t) + Kx(t) = Bu(t) + d, \\; x(t) ∈ \\mathcal{X}, \\; u(t) ∈ \\mathcal{U} \\; \\forall t.
```
Comment (schillic) [inline src/systems.jl:1950]:
```suggestion
    Mx''(t) + Cx'(t) + Kx(t) = Bu(t) + d, \\; x(t) ∈ \\mathcal{X}, \\; u(t) ∈ \\mathcal{U} \\; \\forall t.
```

File: src/systems.jl [hunk 5]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
+        stateset(s::$Z) = s.X
+        inputset(s::$Z) = s.U
+    end
+    for T in [Z, Type{<:eval(Z)}]
+        @eval begin
+            islinear(::$T) = true
+            isaffine(::$T) = true
+            ispolynomial(::$T) = false
+            isnoisy(::$T) = false
+            iscontrolled(::$T) = false
```
Comment (schillic) [inline src/systems.jl:1938]:
```suggestion
            iscontrolled(::$T) = true
```
IIRC, `SecondOrderConstrainedLinearControlContinuousSystem` should return `true` here(?).

File: src/systems.jl [hunk 6]
```diff
@@ -1695,3 +1695,345 @@ with domain constraints of the form:
...
```
Comment (schillic) [inline src/systems.jl:2035]:
```suggestion
            iscontrolled(::$T) = true
```
same for `SecondOrderConstrainedAffineControlContinuousSystem`(?)

**Review by schillic — APPROVED** [review]
LGTM

**Comment by mforets** [comment]
thanks for the review.

**Pull request merged**